### PR TITLE
Allow users to modify their own events (date & address)

### PR DESCRIPTION
**Pull request by tmb**

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -107,4 +116,13 @@ def event_params
         :point_of_contact_id
       )
     end
+
+    def user_event_params
+      params.require(:event).permit(
```
Comment (lachlanjc) [inline app/controllers/events_controller.rb:112]: Should be all on one line

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -107,4 +116,13 @@ def event_params
...
+          :start,
+          :end,
+          :address
+        )
+    end
 end
+
```
Comment (lachlanjc) [inline app/controllers/events_controller.rb]: Extra line here

File: app/views/events/show.html.erb [hunk 3]
```diff
@@ -2,19 +2,21 @@
   <span style="font-size: 3rem;">
     <%= @event.name %>
   </span>
-  <% if admin_signed_in? %>
+
```
Comment (lachlanjc) [inline app/views/events/show.html.erb:5]: Fix indentation here

File: app/controllers/events_controller.rb [hunk 4]
```diff
@@ -54,12 +54,21 @@ def edit
   def update
     authorize @event
 
-    if @event.update(event_params)
-      flash[:success] = 'Event was successfully updated.'
-      redirect_to @event
+    if current_user.admin?
+      if @event.update(event_params)
```
Comment (lachlanjc) [inline app/controllers/events_controller.rb]: This is rather inelegant. Could avoid the repeating through `@event.update(current_user.admin? ? event_params : user_event_params)`

**Review by lachlanjc — CHANGES_REQUESTED** [review]
Nice work here, let’s clean this up before merging though

File: app/controllers/events_controller.rb (hunk 4 above)
Comment (tmb) [reply app/controllers/events_controller.rb]: ah, thanks for this. I figured there was some less ugly way to do it 👍

Comment (tmb) [reply app/controllers/events_controller.rb]: ok also that's so cool you can do ternary as method params omg

File: app/controllers/events_controller.rb [hunk 5]
```diff
@@ -54,12 +54,12 @@ def edit
...
     authorize @event
 
-    if @event.update(event_params)
+    if @event.update(current_user.admin? ? event_params : user_event_params)
       flash[:success] = 'Event was successfully updated.'
       redirect_to @event
     else
       render :edit
-    end
+    end
```
Comment (maxwofford) [inline app/controllers/events_controller.rb]: Trailing whitespace – your editor might have a way to automatically remove these

**Comment by tmb** [comment]
merging in!

**Pull request merged**